### PR TITLE
chore(docs): Add ADR summarizing l10n / i18n improvements RFC

### PR DESCRIPTION
**Pull request by LZoog**
Because:
* We've had discussions over l10n improvements that should be made and want to capture major decisions in an ADR

This commit:
* Creates said ADR

Closes [FXA-5884](https://mozilla-hub.atlassian.net/browse/FXA-5884)

---

I added everyone in fxa-devs that commented in the RFC doc to the deciders list.

All of these decisions have been made except for the last decision regarding submodules/subtree pros/cons, so I'll probably merge this fairly soon after 1-2 approvals.

### CODE REVIEW

File: docs/adr/0032-l10n-i18n-improvements.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+- Deciders: Lauren Zugai, Dan Schomburg, Barry Chen, Vijay Budhram, Wil Clouser
+- Date: 2022-09-22
+
+## Technical Story
+
+[This document](https://docs.google.com/document/d/1TgD7WBLvS4aZ10neLJNk-EYvDR8Kb9GOk2KHxhjVzbk/) was a RFC from 2022-07-09 to 2022-09-19 and contains Jira links, more information, and more context.
+
+## Context and Problem Statement
+
+As Firefox Accounts has migrated from gettext to Fluent over the last 2-3 years, several questions around l10n, including our infrastructure (i18n) and consistency amongst our various packages, as well as discussions around general improvements have surfaced.
```
Comment (clouserw) [inline docs/adr/0032-l10n-i18n-improvements.md:13]: "i18n" is out of place here.  Maybe reword to:

> ...several questions around L10n, including infrastructure, package inconsistencies, and general improvements have surfaced.

File: docs/adr/0032-l10n-i18n-improvements.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+When we use components from `fxa-react` in another package, we have to add the FTL ID and string into the package using the component rather than in `fxa-react`. This can be easily forgotten or lead to duplication of IDs/messages across our packages.
+
+Instead, we can concat files from `fxa-react` into the existing React bundles. Additionally, to help with consistency and maintainability across the board, we'll create a "global" FTL file with our license and branding to be shared across our packages to be used first to build our bundles.
+
+### Other considered options
+
+We could create a new bundle for `fxa-react` similar to other packages. However, we'd need to ship the outputted file with our other packages which would create another network request and more importantly we'd have to reconfigure our Fluent set up to check for translations in two different bundles.
+
+## Decision outcome 4: Clone `fxa-content-server-l10n` once, instead of per-package
```
Comment (clouserw) [inline docs/adr/0032-l10n-i18n-improvements.md:91]: I don't have a strong opinion on the method, but I'm excited not to have to clone this for every package.

File: docs/adr/0032-l10n-i18n-improvements.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+- Allowing `fxa-react` to manage its own l10n
+- More straightforward l10n in emails
+
+## Decision outcome 1: Use consistent FTL paths, names, and setup across packages
+
+`en-US` was FxA's default locale until early 2021 when the l10n team recommended we switch to `en`. `fxa-settings` still contains `en-US.ftl` files that will be updated to `en.ftl`.
+
+Historically, we did not deploy our `en` or `en-US` Fluent bundles, but we have since May 2022 to support any missing fallback text and for consistency across locales and environments.
+
+### `fxa-payments-server` setup changes
```
Comment (clouserw) [inline docs/adr/0032-l10n-i18n-improvements.md:33]: I'm confused by the hierarchy of the document.  You have a decision outcome as an `<h2>` above this subject, but the rest of the doc has the subjects above the decision outcomes and I think inconsistent h1/h2/h3 but that might just be the ordering.  It's probably worth revisiting that to help clarify the ideas here.

Comment (LZoog) [reply docs/adr/0032-l10n-i18n-improvements.md:33]: Yeah, good point, I'll revisit this. The typical ADR format felt a little awkward for this since there were a lot of things to cover and it wasn't a 1:1 with the RFC. I knocked this down from 6-7 "decisions" made to 4, placing some stuff under others.

File: docs/adr/0032-l10n-i18n-improvements.md (hunk 2 above)
Comment (LZoog) [reply docs/adr/0032-l10n-i18n-improvements.md:91]: If someone feels strongly that we should try out subtrees I would be open to it, mostly because `git-subtrees-split` and `git-subtree-merge` seem interesting, but otherwise, it seems like it could be a lot of effort to essentially completely rewrite our script that we'll have to maintain either way. Cloning is straightforward 🤷‍♀️

File: docs/adr/0032-l10n-i18n-improvements.md (hunk 3 above)
Comment (LZoog) [reply docs/adr/0032-l10n-i18n-improvements.md:33]: I pushed an update here.

File: docs/adr/0032-l10n-i18n-improvements.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+- Deciders: Lauren Zugai, Dan Schomburg, Barry Chen, Vijay Budhram, Wil Clouser
+- Date: 2022-09-22
+
+## Technical Story
+
+[This document](https://docs.google.com/document/d/1TgD7WBLvS4aZ10neLJNk-EYvDR8Kb9GOk2KHxhjVzbk/) was a RFC from 2022-07-09 to 2022-09-19 and contains Jira links, more information, and more context.
+
+## Context and Problem Statement
+
+As Firefox Accounts has migrated from gettext to Fluent over the last 2-3 years, several questions several questions around l10n, including infrastructure, package inconsistencies, and general improvements have surfaced.
```
Comment (chenba) [inline docs/adr/0032-l10n-i18n-improvements.md:13]: I see 'several questions' twice.

File: docs/adr/0032-l10n-i18n-improvements.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+## Decision outcome 2: continue using fallback text and set up better testing strategies
+
+In theory, Fluent should always insert translated strings found in its bundles for us, including English strings. Because of this, we considered dropping our fallback text in favor of relying on Fluent.
+
+### Pros of dropping fallback text
+
+- Better maintainability and guarantee of consistency due to not needing duplicate strings (one for fallback text and one in an FTL file), in the codebase
+- It might feel more i18n-friendly to depend on Fluent instead of prioritizing English speaking users by explicitly setting English fallback text
+
+### Cons of dropping fallback text
```
Comment (chenba) [inline docs/adr/0032-l10n-i18n-improvements.md:56]: In theory it is possible that the browser won't be able to load the fluent bundle files after loading the page successfully, at which point the fallback text would be displayed.

File: docs/adr/0032-l10n-i18n-improvements.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+We will create a helper function to send into our templates to more easily grab messages from Fluent bundles for things like alt text that otherwise require additional complexities due to our MJML and `@fluent/dom` set up. We can also automatically apply `data-l10n-args`.
+
+We test our email templates in an enormous, single test file of email strings. We will adjust our email test set up and separate these out per component for better maintainability and easier testing.
+
+## Decision outcome 3: concat `fxa-react` FTL files to other packages, use a global branding file
+
+When we use components from `fxa-react` in another package, we have to add the FTL ID and string into the package using the component rather than in `fxa-react`. This can be easily forgotten or lead to duplication of IDs/messages across our packages.
+
+Instead, we can concat files from `fxa-react` into the existing React bundles. Additionally, to help with consistency and maintainability across the board, we'll create a "global" FTL file with our license and branding to be shared across our packages to be used first to build our bundles.
```
Comment (chenba) [inline docs/adr/0032-l10n-i18n-improvements.md:85]: Did you mean "Fluent bundles" instead of "React bundles"?

File: docs/adr/0032-l10n-i18n-improvements.md (hunk 5 above)
Comment (LZoog) [reply docs/adr/0032-l10n-i18n-improvements.md:56]: 👍 Added another sentence to one of the cons.

File: docs/adr/0032-l10n-i18n-improvements.md (hunk 6 above)
Comment (LZoog) [reply docs/adr/0032-l10n-i18n-improvements.md:85]: I did, thanks! 🙇‍♀️

File: docs/adr/0032-l10n-i18n-improvements.md (hunk 2 above)
Comment (dschom) [reply docs/adr/0032-l10n-i18n-improvements.md:91]: I initially suggested submodules, but no longer recommend this after seeing the comments in the your working document. After thoroughly working through the pros and cons, modifying the existing clone script seems the least disruptive for the team.

**Review by dschom — APPROVED** [review]
:+1:
